src/mapnode.cpp

```diff
@@ -779,15 +772,10 @@ void MapNode::deSerializeBulk(std::istream &is, int version,
 			|| params_width != 2)
 		FATAL_ERROR("Deserialize bulk node data error");
 
-	// Uncompress or read data
-	u32 len = nodecount * (content_width + params_width);
-	std::ostringstream os(std::ios_base::binary);
-	decompressZlib(is, os);
-	std::string s = os.str();
-	if(s.size() != len)
-		throw SerializationError("deSerializeBulkNodes: "
-				"decompress resulted in invalid size");
-	const u8 *databuf = reinterpret_cast<const u8*>(s.c_str());
+	// read data
```

Comment:
We have a `version` parameter. The decompression of the old version is NOT the MapBlock's task. Instead, it should be done here like previously to keep the responsibilities of each function sane.

Reply:
Well... The mapblock is now compressed as a whole (to get better compression, better use of the dictionary, and better performance)

Reply:
Yes, for versions 29+ that's true. However, for older ones it would be a cleaner call structure (IMO) when the MapNode decompression is done in MapNode and not MapBlock.

Reply:
I actually think it's better the way it is now. Node know how to (de)serialize themselves. But all (de)compression logic is in MapBlock (the thing that actually gets stored in the DB).

Why would decompression not be the MapBlock's task? Before this responsibility was spread, now it's all together, which I think is cleaner.